overlord/booted.go

```diff
@@ -57,22 +60,56 @@ func SyncBoot() error {
 	kernelSnap, _ := bootloader.GetBootVar("snappy_kernel")
```

Comment:
No good reason, well, the bootvars are tightly controlled/written by snapd - however that is not really a good reason. I improved the error handling.